azure-go-aci/main.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"github.com/pulumi/pulumi-azure/sdk/go/azure/containerservice"
+	"github.com/pulumi/pulumi-azure/sdk/go/azure/core"
+	"github.com/pulumi/pulumi-docker/sdk/go/docker"
+	"github.com/pulumi/pulumi/sdk/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		// Create a resource group.
+		resourceGroup, err := core.NewResourceGroup(ctx, "aci-rg", nil)
+		if err != nil {
+			return err
+		}
+
+		// Create a registry.
+		registryArgs := containerservice.RegistryArgs{
+			ResourceGroupName: resourceGroup.Name,
+			AdminEnabled:      pulumi.Bool(true),
+			Sku:               pulumi.String("Premium"),
+		}
+		registry, err := containerservice.NewRegistry(ctx, "registry", &registryArgs)
+		if err != nil {
+			return err
+		}
+
+		// Create the docker image.
+		imageArgs := docker.ImageArgs{
+			ImageName: pulumi.Sprintf("%s/mynodeapp:v1.0.0", registry.LoginServer),
+			Build: docker.DockerBuildArgs{
+				Context: pulumi.String("./app"),
+			},
+			Registry: &docker.ImageRegistryArgs{
+				Server:   registry.LoginServer,
+				Username: registry.AdminUsername,
+				Password: registry.AdminPassword,
+			},
+		}
+		image, err := docker.NewImage(ctx, "node-app", &imageArgs)
+		if err != nil {
+			return err
+		}
+
+		// Create a group.
+		credentialArgs := containerservice.GroupImageRegistryCredentialArgs{
+			Server:   registry.LoginServer,
+			Username: registry.AdminUsername,
+			Password: registry.AdminPassword,
+		}
+		portArgs := containerservice.GroupContainerPortArgs{
+			Port:     pulumi.Int(80),
+			Protocol: pulumi.String("TCP"),
+		}
+		containerArgs := containerservice.GroupContainerArgs{
+			Cpu:    pulumi.Float64(0.5),
+			Image:  image.ImageName,
+			Memory: pulumi.Float64(1.5),
+			Name:   pulumi.String("hello-world"),
+			Ports:  containerservice.GroupContainerPortArray([]containerservice.GroupContainerPortInput{portArgs}),
```

Comment:
@pgavlin has pointed out to me that in general - this pattern can always be written as this instead:

```go
containerservice.GroupContainerPortArray({portArgs}),
```

Same applies in a few more places in this file (and in other Go examples we have).

Reply:
When I try `containerservice.GroupContainerPortArray({portArgs})`, Go says it's a syntax error: `syntax error: unexpected {, expecting expression`. Am I missing something?

Reply:
Sorry - typed that wrong - it's:

```go
containerservice.GroupContainerPortArray{portArgs},
```